### PR TITLE
Improve String Extraction in Go Extraction Script for Strings Ending with 0Ah or '\n'

### PR DESCRIPTION
**Pull request by Arker123**
This pull request aims to enhance the go extraction script by enabling the extraction of strings ending with `0A` or `'\n'`
See: #837

### CODE REVIEW

File: floss/results.py [hunk 1]
```diff
@@ -139,7 +139,7 @@ def from_utf8(cls, buf, addr, min_length):
         except UnicodeDecodeError:
             raise ValueError("not utf-8")
 
-        if not decoded_string.isprintable():
+        if not decoded_string.replace("\n", "").isprintable():
```
Comment (mr-tz) [inline floss/results.py:142]: from the docs: "Nonprintable characters are those characters defined in the Unicode character database as “Other” or “Separator”, excepting the ASCII space (0x20) which is considered printable."

I think we should use a different check here or remove as we've already proven this UTF-8 decodes successfully?

Comment (Arker123) [reply floss/results.py:142]: Agreed, let's remove the check.😄

**Review by mr-tz — COMMENTED** [review]
great, can you add a test to show this works as expected now and then let's merge

File: tests/test_language_extract_go.py [hunk 2]
```diff
@@ -124,6 +124,23 @@ def test_mov_lea_mov(request, string, offset, encoding, go_strings):
...
+    "string,offset,encoding,go_strings",
+    [
+        # .text:0000000000481211 48 C7 40 10 19 00 00 00       mov     qword ptr [rax+10h], 19h
+        # .text:0000000000481219 48 8D 0D 71 B6 02 00          lea     rcx, aExpandenvironm ; "ExpandEnvironmentStringsW"
+        # .text:0000000000481220 48 89 48 08                   mov     [rax+8], rcx
+        pytest.param(" markroot jobs done\n", 0xAA68A, StringEncoding.UTF8, "go_strings64"),
+        # .text:0047EACA C7 40 0C 19 00 00 00                          mov     dword ptr [eax+0Ch], 19h
+        # .text:0047EAD1 8D 0D 36 56 4A 00                             lea     ecx, unk_4A5636
+        # .text:0047EAD7 89 48 08                                      mov     [eax+8], ecx
+        pytest.param(" markroot jobs done\n", 0xA23E5, StringEncoding.UTF8, "go_strings32"),
```
Comment (mr-tz) [inline tests/test_language_extract_go.py:137]:
```suggestion
        pytest.param(" markroot jobs done\n", 0xAA68A, StringEncoding.UTF8, "go_strings64"),
        pytest.param(" markroot jobs done\n", 0xA23E5, StringEncoding.UTF8, "go_strings32"),
```

File: tests/test_language_extract_go.py [hunk 3]
```diff
@@ -124,6 +124,23 @@ def test_mov_lea_mov(request, string, offset, encoding, go_strings):
...
+    ],
+)
+def test_strings_with_0A(request, string, offset, encoding, go_strings):
```
Comment (mr-tz) [inline tests/test_language_extract_go.py:140]:
```suggestion
def test_strings_with_newline_0A_character(request, string, offset, encoding, go_strings):
```

File: tests/test_language_extract_go.py [hunk 4]
```diff
@@ -124,6 +124,21 @@ def test_mov_lea_mov(request, string, offset, encoding, go_strings):
...
+@pytest.mark.parametrize(
+    "string,offset,encoding,go_strings",
+    [
+        # 00000000004AB080  20 6E 6F 74 20 66 6F 75  6E 64 20 6D 61 72 6B 72   not found markr
+        # 00000000004AB090  6F 6F 74 20 6A 6F 62 73  20 64 6F 6E 65 0A 20 74  oot jobs done. t
+        pytest.param(" markroot jobs done\n", 0xAA68A, StringEncoding.UTF8, "go_strings64"),
+        # 004A3DE0  66 6F 75 6E 64 20 6D 61  72 6B 72 6F 6F 74 20 6A  found markroot j
+        # 004A3DF0  6F 62 73 20 64 6F 6E 65  0A 20 74 6F 20 75 6E 61  obs done. to una
+        pytest.param(" markroot jobs done\n", 0xA23E5, StringEncoding.UTF8, "go_strings32"),
+    ],
```
Comment (williballenthin) [inline tests/test_language_extract_go.py:136]: this declarative way of asserting test cases is great!

**Review by williballenthin — APPROVED** [review]
looks good to me

**Pull request merged**